src/tasks/installers/operator.ts

```diff
@@ -7,141 +7,173 @@
-            task.title = `${task.title}...done.`
+            await kube.createRoleBindingFrom(roleBinding, flags.chenamespace)
           }
         }
-      },
-      {
-        title: `Create ClusterRoleBinding ${operatorNamespaceEditorClusterRoleBindingName}`,
-        task: async (_ctx: any, task: any) => {
-          const exist = await kube.clusterRoleBindingExist(operatorNamespaceEditorClusterRoleBindingName)
-          if (exist) {
-            task.title = `${task.title}...It already exists.`
+
+        // For Cluster Roles and Cluster Role Bindings use prefix to allow several Che installations
+        const clusterObjectNamePrefix = `${flags.chenamespace}-`
+
+        for (const clusterRole of ctx.clusterRoles as V1ClusterRole[]) {
+          const clusterRoleName = clusterObjectNamePrefix + clusterRole.metadata!.name
+          if (await kube.clusterRoleExist(clusterRoleName)) {
+            if (shouldUpdate) {
+              await kube.replaceClusterRoleFrom(clusterRole, clusterRoleName)
+            }
           } else {
-            await kube.createClusterRoleBinding(operatorNamespaceEditorClusterRoleBindingName, this.operatorServiceAccount, flags.chenamespace, namespaceEditorClusterRoleName)
-            task.title = `${task.title}...done.`
+            await kube.createClusterRoleFrom(clusterRole, clusterRoleName)
           }
         }
-      },
-      {
-        title: `Create RoleBinding ${this.operatorRoleBinding} in namespace ${flags.chenamespace}`,
-        task: async (ctx: any, task: any) => {
-          const exist = await kube.roleBindingExist(this.operatorRoleBinding, flags.chenamespace)
-          if (exist) {
-            task.title = `${task.title}...It already exists.`
+
+        for (const clusterRoleBinding of ctx.clusterRoleBindings as V1ClusterRoleBinding[]) {
+          clusterRoleBinding.metadata!.name = clusterObjectNamePrefix + clusterRoleBinding.metadata!.name
+          clusterRoleBinding.roleRef.name = clusterObjectNamePrefix + clusterRoleBinding.roleRef.name
+          for (const subj of clusterRoleBinding.subjects || []) {
+            subj.namespace = flags.chenamespace
+          }
+          if (await kube.clusterRoleBindingExist(clusterRoleBinding.metadata!.name)) {
+            if (shouldUpdate) {
+              await kube.replaceClusterRoleBindingFrom(clusterRoleBinding)
+            }
           } else {
-            const yamlFilePath = ctx.resourcesPath + 'role_binding.yaml'
-            await kube.createRoleBindingFromFile(yamlFilePath, flags.chenamespace)
-            task.title = `${task.title}...done.`
+            await kube.createClusterRoleBindingFrom(clusterRoleBinding)
           }
         }
-      },
+
+        task.title = `${task.title}...done.`
+      }
+    }
+  }
+
+  /**
+   * Returns tasks list which perform preflight platform checks.
+   */
+  deployTasks(flags: any, command: Command): Listr {
```

Comment:
Maybe public methods should be upper?

Reply:
To me they should be in their place. I see no point in separating a public method with its helper jst because the helper is private. That way it is easier to read code.